### DISCUSSION TITLE
Angular 14 compatibility

### DISCUSSION THREAD

**Discussion by mazinPos** [boris-jenicek/ngx-awesome-popup#40] [Q&A]
When I try to use this library to show a pop-up upon button click. I get a [loading indicator](https://ibb.co/4t16xwD) that remains forever.

I noticed from the [tutorial provided](https://costlydeveloper.github.io/ngx-awesome-popup/#/)  that it mentions Angular 13.

Is this library compatible with Angular 14? if yes, what is causing the issue I am facing?

Your guidance is appreciated

**Comment by boris-jenicek** [comment]
Sorry for waiting so long, it works with angular 14 and future versions.

There are two options to achieve that:

**1. Invoke a dialog without a loader**
```ts
dialogPopup.setConfig({
            width     : '500px',
            layoutType: DialogLayoutDisplay.NONE, // SUCCESS | INFO | NONE | DANGER | WARNING
            displayLoader:  false
        });
```

**2. Invoke dialog with a loader and close it from the dialog component itself.**

```ts
dialogPopup.setConfig({
            width     : '500px',
            layoutType: DialogLayoutDisplay.NONE, // SUCCESS | INFO | NONE | DANGER | WARNING
            displayLoader: true
        });
```
from a component that is invoked in dialog, you need to call DI(dependency injection) dialogBelonging look here https://stackblitz.com/edit/minimal-dialog-setup?file=src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fany-angular%2Fany-angular.component.ts
and when you want to close the loader call event to close the loader like this: 

```ts
this.dialogBelonging.eventsController.closeLoader();
```
